mixer/pkg/runtime/handler/env_test.go

```diff
@@ -57,28 +57,38 @@ func TestEnv(t *testing.T) {
 
 		wg := sync.WaitGroup{}
 		wg.Add(1)
-		env.ScheduleWork(func() {
+		e.ScheduleWork(func() {
 			wg.Done()
 		})
 
 		wg.Add(1)
-		env.ScheduleDaemon(func() {
+		e.ScheduleDaemon(func() {
 			wg.Done()
 		})
 
-		env.ScheduleWork(func() {
+		e.ScheduleWork(func() {
 			panic("bye!")
 		})
 
-		env.ScheduleDaemon(func() {
+		e.ScheduleDaemon(func() {
 			panic("bye!")
 		})
 
+		if got, want := e.(env).hasStrayWorkers(), true; got != want {
+			t.Errorf("hasStrayWorkers() => %t; wanted %t", got, want)
+		}
+
+		e.(env).reportStrayWorkers()
+
 		wg.Wait()
 
 		// hack to give time for the panic to 'take hold' if it doesn't get recovered properly
 		time.Sleep(200 * time.Millisecond)
```

Comment:
nit: I'd use channels to communicate with the daemon instead of relying on timing. I think this might be OK now, but golang might decide to interrupt between lines 75 and 76 in the future.

Reply:
moved the check inside the goroutines. just wanted this to validate + coverage.